src/__tests__/auth.spec.js

```diff
@@ -1,5 +1,264 @@
+const UserModel = require('../models/user');
+
+const mockUser = {
+  sub: '12345678',
+  name: 'John Doe',
+  given_name: 'John',
+  family_name: 'Doe',
+  picture: 'https://lh3.googleusercontent.com/a-/AOh1',
+  email: 'john.doe@gmail.com',
+  email_verified: true,
+  locale: 'en-GB',
+};
+
+beforeAll(async () => {
+  await db();
+});
+
+afterAll(async (drop = false) => {
+  drop && (await mongoose.connection.dropDatabase());
+  await mongoose.disconnect();
+  await mongoose.connection.close();
+});
+
+let redirectUri = null;
+
+describe('Google Auth Endpoints', () => {
+  const cookiesAgent = supertest.agent(app);
+
+  afterAll(async () => {
+    await UserModel.deleteMany();
+  });
+
+  describe('GET /api/auth/google', () => {
+    it('Redirects to google authorization page', (done) => {
+      req
+        .get('/api/auth/google')
+        .expect(302)
+        .expect('location', /google\.com/)
+        .end(done);
+    });
+
+    it('Redirects with correct scope and credentials', async () => {
+      const res = await req.get('/api/auth/google');
+      const location = res.header.location;
+
+      expect(location).not.toBeNull();
+
+      const uri = new URL(location);
+      const scope = uri.searchParams.get('scope')?.split(' ') ?? [];
+      const redirectTo = uri.searchParams.get('redirect_uri') ?? '';
+      const client_id = uri.searchParams.get('client_id') ?? '';
+
+      expect(scope).toEqual(
+        expect.arrayContaining(['openid', 'email', 'profile'])
+      );
+      // expect(redirectTo).toMatch(URL_REGEX);
+      expect(client_id.length).toBeGreaterThan(10);
+
+      if (redirectTo) redirectUri = new URL(redirectTo);
+    });
+  });
+
+  describe(`GET REDIRECT_URI`, () => {
+    it('Redirects to home without cookie for incorrect credentials', async () => {
+      expect(redirectUri).not.toBeNull();
+      const res = await req.get(redirectUri.pathname);
+      expect(res.status).toBe(302);
+      expect(res.header['set-cookie']).not.toBeDefined();
+    });
+
+    it('Redirects to home with a valid JWT cookie for correct credentials', async () => {
+      expect(redirectUri).not.toBeNull();
+
+      const res = await runInPatchedServer(
+        async () => await cookiesAgent.get(getLoginURL(redirectUri.pathname))
+      );
+
+      expect(res.status).toBe(302);
+      expect(res.header['set-cookie']).toBeDefined();
+
+      const [cookies] = parseCookies(res.header['set-cookie']);
+
+      const auth_cookie = getJWTCookie(cookies);
+
+      const iat = auth_cookie.iat;
+
+      expect(iat).toBeLessThanOrEqual(Date.now() / 1000);
+
+      const expected = {
+        // name: mockUser.name,
+        email: mockUser.email,
+        providerId: `google-${mockUser.sub}`,
+        // avatar: mockUser.picture,
+        exp: iat + 14 * 24 * 3600,
+        iat: expect.any(Number),
+      };
+
+      expect(auth_cookie).toEqual(expect.objectContaining(expected));
+    });
   });
 });
+
+async function runInPatchedServer(cb) {
```

Comment:
Naming can be clearer

Reply:
changed it to `runGoogleRequestInPatchedServer`